python/lsst/cp/pipe/cpTask.py

```diff
@@ -0,0 +1,594 @@
+        self.config.validate()
+        self.config.freeze()
+
+        # One of these for each pexConfig.ConfigurableField
+        self.makeSubtask("fe55")
+        self.makeSubtask("readNoise")
+        self.makeSubtask("brightPixels")
+        self.makeSubtask("darkPixels")
+        self.makeSubtask("traps")
+        self.makeSubtask("cte")
+        self.makeSubtask("flatPair")
+        self.makeSubtask("ptc")
+
+        # Do we want to switch to the better-named logger below or use the one we get from cmdLineTask?
+        # e.g. using self.log = lsstLog.Log.getLogger("cp.pipe.cpTask")?
+
+    def _getMaskFiles(self, path, ccd):
+        """Get all available eotest mask files for a given ccd.
+
+        Each stage of the processing generates more mask files, so this allows each to be picked up
+        as more and more tests run, and saves having to have clever logic for if some tasks fail.
+        Parameters
+        ----------
+        path : string
+            Path on which to find the mask files
+        ccd : string/int
+            Name/identifier of the CCD
+        Returns
+        -------
+        maskFiles : list/tuple
+            List of mask files, or an empty tuple if none are found
+        """
+        pattern = '*' + str(ccd) + '*mask*'  # the cast to str supports obs_auxTel
+        maskFiles = glob.glob(os.path.join(path, pattern))
+        return maskFiles if len(maskFiles) > 0 else ()  # eotest wants an empty tuple here
+
+    def _cleanupEotest(self, path):
+        """Delete all the medianed files left behind after eotest has run.
+
+        Running eotest generates a lot of interim medianed files, so this just cleans them up.
+        Parameters
+        ----------
+        path : string
+           Path on which to delete all the eotest medianed files.
+        """
+        for filename in glob.glob(os.path.join(path, '*median*')):
+            os.remove(filename)
+
+    def _gainPropSetToDict(self, pSet):
+        """Translator for the persisted gain values.
+
+        eotest wants an {amp: gain} dictionary with integer keys for the amps.
+        When we persist dafBase.propertySets these only take strings for keys,
+        so the provided .toDict() method won't do, so here we provide an extra
+        layer of translation to keep eotest happy.
+
+        Parameters
+        ----------
+        pSet : daf.base.PropertySet
+           PropertySet to be translated to an integer-keyed dictionary
+        """
+        return {int(amp): gain for amp, gain in pSet.toDict().items()}
+
+    def testMethod(self):  # TODO: XXX remove this whole method, just useful for debugging/dev work
+        """TODO: Remove this docstring that only exists to remove flake8 error."""
+        self.log.warn("Test warning message")
+        print(self.config.requireAllEOTests)
+
+    # @pipeBase.timeMethod #xxx re-include this
+    def runEotestDirect(self, butler, run=None):
```

Comment:
This is a "super method": it's trying to do too much. In general, a method should do one thing, but this is trying to do the dispatch for each of the individual parts. That means that if you want to subclass and override this method, you need to copy a lot of functional code, which is bad.

I suggest turning this method into a collection of `if` statements that call other methods to do the dispatch:

    if self.config.doFe55:
        self.runFe55(run, ...)
    if self.config.doReadNoise:
        self.runReadNoise(run, ...)

There's a lot of code duplication between the various dispatches, which you should factor out.

You should also ask yourself if this approach is really a good idea --- do you actually run each of the individual parts on every file? But this isn't the `run` method. Where *is* the `run` method? Without it, this `CmdLineTask` won't do *anything* when run. And it looks like you're not actually iterating over individual files (more like all files of a certain kind), so `CmdLineTask` (at least, as set up by default) isn't what you want. This code (as it currently stands) is in no-man's land.